lib/grape-swagger.rb

```diff
@@ -49,6 +40,26 @@ def add_swagger_documentation(options = {})
 
       private
 
+      def version_for(options)
+        options[:version] = version if version
+      end
+
+      def combine_routes(app, doc_klass)
+        app.routes.each do |route|
+          route_path = route.route_path
```

Comment:
I like that you created a new method for that. But unfortunately it's still hard to understand. Maybe you can generate some more methods to make the code more readable. E.g.:
A separate questionmark method for `route_path.split(/^.*?#{route.route_prefix.to_s}/).last`, `route_match.match('\/([\w|-]*?)[\.\/\(]') || route_match.match('\/([\w|-]*)$')` and `doc_klass.hide_documentation_path && route.route_path.match(/#{doc_klass.mount_path}($|\/|\(\.)/)`

With that, the could would be better readable and the method lengths will increase.